### PR TITLE
[ANCHOR-369] Migrate SEP-10 to use clients configuration

### PR DESCRIPTION
**Pull request by lijamie98**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What
Migrate SEP-10 to use clients configuration 

### Why
Feature implementation

### CODE REVIEW

File: platform/src/main/resources/config/anchor-config-default-values.yaml [hunk 1]
```diff
@@ -332,17 +323,14 @@ sep10:
...
+  ## @param: require_known_custodial_account
   ## @type: bool
-  ## Whether to require authenticating clients to be in the omnibusAccountList
+  ## Whether to require authenticating clients to be in the list of known custodial accounts.
+  #
+  # If the flag is set to true, the client must be one of the custodial clients defined in the clients section
+  # of this configuration file.
   #
-  require_known_omnibus_account: false
+  require_known_custodial_account: false
```
Comment (JakeUrban) [inline platform/src/main/resources/config/anchor-config-default-values.yaml:333]: This is a breaking change, right?

Comment (lijamie98) [reply platform/src/main/resources/config/anchor-config-default-values.yaml:333]: Yes... I assume the impact would be minimum if no anchors on version `2.x` are using this flag. 

WDYT?

Comment (JakeUrban) [reply platform/src/main/resources/config/anchor-config-default-values.yaml:333]: Don't think any custodial wallets have integrated with any of the SEP-24 anchors live on the platform yet, so we may be ok to make this change.

File: core/src/main/java/org/stellar/anchor/config/Sep10Config.java [hunk 2]
```diff
@@ -3,23 +3,75 @@
...
+  /**
+   * The timeout in seconds of the authenticated JSON Web Token. An expired JWT will be rejected.
+   * This is the timeout period after the client has authenticated.
+   *
+   * @return the timeout in seconds of the authenticated JSON Web Token
+   */
   Integer getJwtTimeout();
 
-  List<String> getClientAttributionDenyList();
+  boolean isRequireKnownOmnibusAccount();
```
Comment (JakeUrban) [inline core/src/main/java/org/stellar/anchor/config/Sep10Config.java:62]: Don't we need to change these methods to match the config?

Comment (lijamie98) [reply core/src/main/java/org/stellar/anchor/config/Sep10Config.java]: Ah... missed it in PR merge. It is now fixed.

**Review by philipliu — APPROVED** [review]
LGTM

**Review by Ifropc — APPROVED** [review]

File: core/src/main/java/org/stellar/anchor/config/Sep10Config.java [hunk 3]
```diff
@@ -3,23 +3,75 @@
...
+  /**
+   * Set if the client attribution is required. Client Attribution requires clients to verify their
+   * identity by passing a domain in the challenge transaction request and signing the challenge
+   * with the ``SIGNING_KEY`` on that domain's SEP-1 stellar.toml. See the SEP-10 section `Verifying
+   * Client Application Identity` for more information, refer to <a
+   * href="https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0010.md#verifying-client-application-identity">SEP-10</a>.
+   *
+   * @return true if the client attribution is required
+   */
   boolean isClientAttributionRequired();
```
Comment (JakeUrban) [inline core/src/main/java/org/stellar/anchor/config/Sep10Config.java:36]: Lets add that `isClientAttributionRequired()` is only relevant for noncustodial wallets, and `isKnownCustodialAccountRequired()` is reciprocal method for custodial wallets.

File: core/src/main/java/org/stellar/anchor/config/Sep10Config.java [hunk 4]
```diff
@@ -3,23 +3,75 @@
...
+  /**
+   * The timeout in seconds of the authenticated JSON Web Token. An expired JWT will be rejected.
+   * This is the timeout period after the client has authenticated.
+   *
+   * @return the timeout in seconds of the authenticated JSON Web Token
+   */
   Integer getJwtTimeout();
 
-  List<String> getClientAttributionDenyList();
+  boolean isRequireKnownCustodialAccount();
```
Comment (JakeUrban) [inline core/src/main/java/org/stellar/anchor/config/Sep10Config.java:62]: nit: can we do `isKnownCustodialAccountRequired()`?

File: platform/src/main/resources/config/anchor-config-default-values.yaml (hunk 1 above)
Comment (JakeUrban) [reply platform/src/main/resources/config/anchor-config-default-values.yaml:333]: nit: can we match the phrasing of `client_attribution_required` and use `known_custodial_account_required`?

Comment (lijamie98) [reply platform/src/main/resources/config/anchor-config-default-values.yaml:333]: `custodial_account_required`? WDYT?

**Pull request merged**